cosmovisor/args_test.go

```diff
@@ -5,18 +5,73 @@ import (
 	"os"
 	"path/filepath"
 	"testing"
+	"time"
 
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
 )
 
 type argsTestSuite struct {
 	suite.Suite
+
+	envVars []string
```

Comment:
how about using a map instead of a list? As it is now, it's not very readable - one need to remember the order of env variables.

Reply:
This list is needed no matter what because the tests need to be able to clear out all environment variables involved.

At one point, I had `clearEnv` returning a map and `setEnv` taking in a map, but I didn't like the verbosity and redundancy that was then needed for the test case definitions in `TestGetConfigFromEnv`. I thought about several other solutions, but they all felt like they would be overkill for what is needed.

In the end, I decided that, since it's unit tests and specific to just this file, the `setEnv` helper func could get away with being set up as it is.

Reply:
How about having a struct? This way you will be able to have a short initialization in tests, eg:
```
envVals: CosmovisorEnv{absPath, "testname", "true", "false", "true", "303", "1"}
```
And it will be easy to inspect what each true / false means.

Reply:
I had thought about that, but held off because I didn't think it was that much more helpful that looking at a slice of strings and didn't really want to over-engineer stuff for unit tests. But you're right, it's not the same as looking at a slice of strings, and does make things easier.

I'll make that change.

Reply:
Updated those unit tests to use a `cosmovisorEnv` struct.

I still don't think it was the right move, though. It's now significantly more difficult to maintain the tests when changes are made (specially adding a new environment variable).

Using the struct:
1. Add new field to struct.
1. Add new case to setter.
1. Add new entry to mapper.
1. Update all tests that create a `cosmovisorEnv` to include a value for the new field.
1. Update the couple tests that might have broken due to the actual change.
1. Add new unit tests.

Using the string slices:
1. Add new env var name to `s.envVars`.
1. Update the couple tests that might have broken due to the actual change.
1. Add new unit test cases.

To me, the difficulty in readability of the string slice way is significantly less of a problem than the added difficulty in maintaining the tests. Plus, the more infrastructure you create in order to do unit tests, the more likely you are to introduce conditions where tests should fail, but don't.